app/index.mjs

```diff
@@ -0,0 +1,473 @@
+  };
+
+  const [account] = await jsonrpc({ method: "eth_requestAccounts" });
+  const [chainId, bootstrap, factory] = await Promise.all([
+    jsonrpc({ method: "eth_chainId" }),
+    deployedBootstrap(),
+    deployedFactory(),
+  ]);
+
+  wallet.innerText = `${checksumAddress(account)}@${BigInt(chainId)}`;
+
+  if (bootstrap !== null) {
+    b.status.innerText = "Deployed";
+  } else {
+    b.status.innerText = "Not Deployed";
+    b.deploy.disabled = factory !== null;
+  }
+
+  if (factory !== null) {
+    f.status.innerText = "Deployed";
+  } else {
+    f.status.innerText = "Not Deployed";
+    f.deploy.disabled = bootstrap === null;
+  }
+}
+
+function isAddress(value) {
+  return `${value}`.match(/^0x[0-9A-Fa-f]{40}$/);
+}
+
+async function deployedBootstrap(newValue) {
+  const chainId = await jsonrpc({ method: "eth_chainId" });
+  const key = `bootstrap:${chainId}`;
+
+  const value = window.localStorage.getItem(key);
+  if (isAddress(newValue)) {
+    window.localStorage.setItem(key, newValue);
+  }
+
+  if (isAddress(value)) {
+    return value;
+  } else {
+    return null;
+  }
+}
+
+async function deployedFactory() {
+  const address = "0xC0DEb853af168215879d284cc8B4d0A645fA9b0E";
+  const code = await jsonrpc({
+    method: "eth_getCode",
+    params: [address, "latest"],
+  });
+  if (code === "0x60203d3d3582360380843d373d34f5806019573d813d933efd5b3d52f3") {
+    return address;
+  } else {
+    return null;
+  }
+}
+
+function burnerAccount() {
+  const storedKey = window.localStorage.getItem("burner") ?? "";
+  const accountExists = storedKey.match(/^[0-9a-f]{64}$/);
+  const privateKey = accountExists ? hexToBytes(storedKey) : randomBytes(32);
+  if (!accountExists) {
+    window.localStorage.setItem("burner", bytesToHex(privateKey));
+  }
+
+  const publicKey = secp256k1.getPublicKey(privateKey, false);
+  const addressBytes = keccak_256(publicKey.subarray(1)).subarray(12);
+  const address = checksumAddress(bytesToHex(addressBytes));
+  return { address, privateKey };
+}
+
+async function deployBootstrap() {
+  const status = document.querySelector("#bootstrap-status");
+  const deploy = document.querySelector("#bootstrap-deploy");
+
+  status.innerText = "Deploying...";
+  deploy.disabled = true;
+  try {
+    const { bytecode } = BootstrapArtifact;
+    const { status, contractAddress } = await sendTransaction({
+      data: bytecode,
+    });
+    if (status === "0x1") {
+      await deployedBootstrap(contractAddress);
+    }
+  } finally {
+    await updateStatuses();
+  }
+}
+
+async function deployFactory() {
+  const burner = document.querySelector("#burner-address");
+  const notify = document.querySelector("#factory-notify");
+  const status = document.querySelector("#factory-status");
+  const deploy = document.querySelector("#factory-deploy");
+
+  status.innerText = "Deploying...";
+  deploy.disabled = true;
+  try {
+    const bootstrap = await deployedBootstrap();
```

Comment:
Would it make sense to null check here?

Reply:
It _shouldn't_ be necessary, as the button that triggers this handler is only enabled if that value is not null.